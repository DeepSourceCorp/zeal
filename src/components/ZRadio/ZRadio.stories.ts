import '../../assets/css/tailwind.css'
import '../../assets/css/typography.css'
import '../../assets/css/layout.css'
import ZRadio from './ZRadio.vue'

export default {
  title: 'Form Inputs and Controls: Radio Button',
  component: ZRadio,
  excludeStories: /.*Data$/
}

export const DefaultRadio = () => ({
  components: { ZRadio },
  data() {
    return {
      modelValue: ''
    }
  },
  template: `<div class='padded-container'>
            <div class="input-container">
                <z-radio value="female" v-model="modelValue" label="Female"></z-radio>
                <z-radio value="male" v-model="modelValue" label="Male"></z-radio>
            </div>
        </div>`
})

export const CheckedRadioButton = () => ({
  components: { ZRadio },
  data() {
    return {
      modelValue: 'female'
    }
  },
  template: `<div class='padded-container'>
            <div class="input-container">
                <z-radio value="female" v-model="modelValue" label="Female"></z-radio>
                <z-radio value="male" v-model="modelValue" label="Male"></z-radio>
            </div>
        </div>`
})

export const DisabledRadioButton = () => ({
  components: { ZRadio },
  data() {
    return {
      modelValue: 'female'
    }
  },
  template: `<div class='padded-container'>
            <div class="input-container">
                <z-radio value="female" v-model="modelValue" label="Female"></z-radio>
                <z-radio value="male" v-model="modelValue" label="Male" :disabled="true"></z-radio>
            </div>
        </div>`
})

export const DisabledCheckedRadioButton = () => ({
  components: { ZRadio },
  data() {
    return {
      modelValue: 'female'
    }
  },
  template: `<div class='padded-container'>
            <div class="input-container">
                <z-radio value="female" v-model="modelValue" label="Female"></z-radio>
                <z-radio value="male" v-model="modelValue" label="Male"></z-radio>
            </div>
        </div>`
})
