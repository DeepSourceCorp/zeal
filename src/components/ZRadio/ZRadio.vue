<template>
    <label class="z-radio text-vanilla-100 flex items-center"
            :class="`${disabled && 'cursor-not-allowed' || 'cursor-pointer'}`">
        <span class="z-radio--inner rounded-full w-4 h-4 relative cursor-pointer inline-block box-border border-solid border-2"
                :class="[`${isChecked && (disabled && 'border-slate bg-slate' || 'border-juniper bg-juniper') || 'border-slate bg-transparent'}`,
                        `${disabled && 'cursor-not-allowed hover:border-slate' || 'cursor-pointer hover:border-juniper'}`]">
            <span v-if="isChecked" 
                class="w-2 h-2 rounded-full absolute left-50 top-50 transform -translate-x-1/2 -translate-y-1/2"
                :class="`${disabled && 'bg-vanilla-400' || 'bg-vanilla-100'}`">
                </span>
        </span>
        <input type="radio" 
            :checked="isChecked" 
            :value="value" 
            @change="updateInput"
            :disabled="disabled"
            class="opacity-0 absolute h-0 w-0">
        <span class="label ml-2"
              :class="disabled && 'cursor-not-allowed text-slate' || 'cursor-pointer'">{{label}}</span>
    </label>
</template>

<script lang="ts">
import Vue from 'vue'

export default Vue.extend({
  name: 'ZRadio',
  model: {
    prop: "modelValue",
    event: "change"
  },
  props: {
    modelValue: {
        type: String,
        default: ""
    },
    label: {
        type: String,
        required: true
    },
    value: {
        type: String
    },
    disabled: {
        type: Boolean,
        default: false
    }
  },
  computed: {
    isChecked(): boolean {
        return this.modelValue === this.value
    }
  },
  methods: {
      updateInput(): void {
          this.$emit('change',this.value)
      }
  }
})
</script>

