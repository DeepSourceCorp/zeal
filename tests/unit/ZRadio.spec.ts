import ZRadio from '../../src/components/ZRadio'
import Vue from 'vue'

import { mount, Wrapper } from '@vue/test-utils'

describe('Radio Button Component', () => {
  let mountFn: (options?: Record<string, unknown>) => Wrapper<Vue>
  let localVue: typeof Vue

  beforeEach(() => {
    mountFn = (options = {}) => {
      return mount(ZRadio, {
        localVue,
        ...options
      })
    }
  })

  it('renders the default radio component', () => {
    const wrapper = mountFn({
      propsData: {
        label: 'Hello'
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.findAll('input[type="radio"]').length).toEqual(1)
  })

  it('renders the checked radio component', () => {
    const wrapper = mountFn({
      propsData: {
        label: 'Female',
        modelValue: 'female',
        value: 'female'
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.findAll('.z-radio--inner span').length).toEqual(1)
  })
  it('renders the disabled radio component', () => {
    const wrapper = mountFn({
      propsData: {
        label: 'Female',
        value: 'female',
        disabled: true
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.findAll('input[disabled="disabled"]').length).toEqual(1)
  })
  it('renders a checked radio button disabled', () => {
    const wrapper = mountFn({
      propsData: {
        label: 'Female',
        value: 'female',
        disabled: true,
        modelValue: 'female'
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.findAll('input[disabled="disabled"]').length).toEqual(1)
    expect(wrapper.findAll('.z-radio--inner span').length).toEqual(1)
  })
})
