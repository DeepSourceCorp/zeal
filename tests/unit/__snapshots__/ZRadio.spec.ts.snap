// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio Button Component renders a checked radio button disabled 1`] = `<div role="radiogroup" class="z-radio-group"><label class="z-radio text-vanilla-100 flex items-center cursor-not-allowed"><span class="z-radio--inner rounded-full w-4 h-4 relative cursor-pointer inline-block box-border border-solid border-2 border-slate bg-slate cursor-not-allowed hover:border-slate"><span class="w-2 h-2 rounded-full absolute left-50 top-50 transform -translate-x-1/2 -translate-y-1/2 bg-vanilla-400"></span></span> <input type="radio" disabled="disabled" class="opacity-0 absolute h-0 w-0" value="female"> <span class="label ml-2 cursor-not-allowed text-slate">Female</span></label> <label class="z-radio text-vanilla-100 flex items-center cursor-not-allowed"><span class="z-radio--inner rounded-full w-4 h-4 relative cursor-pointer inline-block box-border border-solid border-2 border-slate bg-transparent cursor-not-allowed hover:border-slate"><!----></span> <input type="radio" disabled="disabled" class="opacity-0 absolute h-0 w-0" value="male"> <span class="label ml-2 cursor-not-allowed text-slate">Male</span></label></div>`;

exports[`Radio Button Component renders a radio button with button style like switcher 1`] = `
<div class="padded-container">
  <div class="input-container">
    <div role="radiogroup" class="z-radio-group flex"><label class="flex items-center z-radio-button text-vanilla-100 border-ink-200 last:border-r cursor-pointer"><span class="box-border relative inline-block text-sm font-medium border border-r-0 border-solid cursor-pointer z-radio--inner border-ink-200 bg-ink-200 text-vanilla-100 px-3 py-2"> Female</span> <input type="radio" class="absolute w-0 h-0 opacity-0" value="female"></label> <label class="flex items-center z-radio-button text-vanilla-100 border-ink-200 last:border-r cursor-pointer"><span class="box-border relative inline-block text-sm font-medium border border-r-0 border-solid cursor-pointer z-radio--inner border-ink-200 bg-transparent text-vanilla-400 px-3 py-2"> Male</span> <input type="radio" class="absolute w-0 h-0 opacity-0" value="male"></label></div>
  </div>
</div>
`;

exports[`Radio Button Component renders the checked radio component 1`] = `<div role="radiogroup" class="z-radio-group"><label class="z-radio text-vanilla-100 flex items-center cursor-pointer"><span class="z-radio--inner rounded-full w-4 h-4 relative cursor-pointer inline-block box-border border-solid border-2 border-juniper bg-juniper cursor-pointer hover:border-juniper"><span class="w-2 h-2 rounded-full absolute left-50 top-50 transform -translate-x-1/2 -translate-y-1/2 bg-vanilla-100"></span></span> <input type="radio" class="opacity-0 absolute h-0 w-0" value="female"> <span class="label ml-2 cursor-pointer">Female</span></label> <label class="z-radio text-vanilla-100 flex items-center cursor-pointer"><span class="z-radio--inner rounded-full w-4 h-4 relative cursor-pointer inline-block box-border border-solid border-2 border-slate bg-transparent cursor-pointer hover:border-juniper"><!----></span> <input type="radio" class="opacity-0 absolute h-0 w-0" value="male"> <span class="label ml-2 cursor-pointer">Male</span></label></div>`;

exports[`Radio Button Component renders the default radio component 1`] = `<div role="radiogroup" class="z-radio-group"><label class="z-radio text-vanilla-100 flex items-center cursor-pointer"><span class="z-radio--inner rounded-full w-4 h-4 relative cursor-pointer inline-block box-border border-solid border-2 border-slate bg-transparent cursor-pointer hover:border-juniper"><!----></span> <input type="radio" class="opacity-0 absolute h-0 w-0" value="female"> <span class="label ml-2 cursor-pointer">Female</span></label> <label class="z-radio text-vanilla-100 flex items-center cursor-pointer"><span class="z-radio--inner rounded-full w-4 h-4 relative cursor-pointer inline-block box-border border-solid border-2 border-slate bg-transparent cursor-pointer hover:border-juniper"><!----></span> <input type="radio" class="opacity-0 absolute h-0 w-0" value="male"> <span class="label ml-2 cursor-pointer">Male</span></label></div>`;

exports[`Radio Button Component renders the disabled radio component 1`] = `<div role="radiogroup" class="z-radio-group"><label class="z-radio text-vanilla-100 flex items-center cursor-not-allowed"><span class="z-radio--inner rounded-full w-4 h-4 relative cursor-pointer inline-block box-border border-solid border-2 border-slate bg-transparent cursor-not-allowed hover:border-slate"><!----></span> <input type="radio" disabled="disabled" class="opacity-0 absolute h-0 w-0" value="female"> <span class="label ml-2 cursor-not-allowed text-slate">Female</span></label> <label class="z-radio text-vanilla-100 flex items-center cursor-not-allowed"><span class="z-radio--inner rounded-full w-4 h-4 relative cursor-pointer inline-block box-border border-solid border-2 border-slate bg-transparent cursor-not-allowed hover:border-slate"><!----></span> <input type="radio" disabled="disabled" class="opacity-0 absolute h-0 w-0" value="male"> <span class="label ml-2 cursor-not-allowed text-slate">Male</span></label></div>`;
